quarkus-opentelemetry-exporter-gcp/runtime/src/main/java/io/quarkiverse/opentelemetry/exporter/gcp/runtime/LateBoundSpanProcessor.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkiverse.opentelemetry.exporter.gcp.runtime;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.jboss.logging.Logger;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+
+public class LateBoundSpanProcessor implements SpanProcessor {
```

Comment:
I wonder if we could use the LateBoundSpanProcessor that is already present on the main OTel extension...

Reply:
Which dependency is that?

Reply:
Do you refer to [`LateBoundBatchSpanProcessor`](https://github.com/quarkiverse/quarkus-opentelemetry-exporter/blob/main/quarkus-opentelemetry-exporter-jaeger/runtime/src/main/java/io/quarkiverse/opentelemetry/exporter/jaeger/runtime/LateBoundBatchSpanProcessor.java) or to `LateBoundBatchSpanProcessor` in `quarkus-opentelemetry` library?

Both of them have `BatchSpanProcessor` as a delegate and we need `SpanProcessor` iface to accept `SimpleSpanProcessor`.

Reply:
Right... it cannot be used because in here the simple is also used.
All good then.